### PR TITLE
Fixup after fixing json

### PR DESCRIPTION
**Pull request by rock3r**
This PR fixes the JSON parsing after actually fixing the JSON on firebase; it also fixes data sorting in the schedule to be more resilient.

### CODE REVIEW

File: app/src/main/java/net/squanchy/schedule/ScheduleService.java [hunk 1]
```diff
@@ -35,12 +42,41 @@
...
                 .withLatestFrom(daysObservable, combineSessionsById())
+                .map(sortPagesByDate())
+                .map(sortEventsByStartDate())
                 .subscribeOn(Schedulers.io());
     }
 
-    private Function<List<Event>, HashMap<String, List<Event>>> mapEventsToDays() {
+    private Function<Schedule, Schedule> sortPagesByDate() {
+        return schedule -> {
+            ArrayList<SchedulePage> sortedPages = new ArrayList<>(schedule.pages());
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/ScheduleService.java:54]: No more mutating stuff for sorting! @fourlastor will be happy

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.java [hunk 2]
```diff
@@ -27,15 +27,16 @@ public static Event create(
...
                 .id(eventId)
                 .numericId(numericEventId)
                 .dayId(dayId)
                 .startTime(startTime)
                 .endTime(endTime)
                 .title(title)
-                .place(Optional.fromNullable(place))
-                .placeVisibility(place.isEmpty() ? View.GONE : View.VISIBLE)
+                .place(placeOptional)
+                .placeVisibility(placeOptional.map(placeValue -> View.VISIBLE).or(View.GONE))
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/domain/view/Event.java:39]: This fixes a NPE crash (place is supposed to be optional in the model, but it's not in _our_ models, so here's the issue)

**Review by tiwiz — APPROVED** [review]

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.java (hunk 2 above)
Comment (fourlastor) [inline app/src/main/java/net/squanchy/schedule/domain/view/Event.java:39]: I'd use an if on `isPresent` here

File: app/src/main/java/net/squanchy/schedule/ScheduleService.java (hunk 1 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/schedule/ScheduleService.java:54]: Sigh

**Review by fourlastor — COMMENTED** [review]
One comment

**Pull request merged**